lib/_stream_readable.js

```diff
@@ -695,17 +713,21 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
     debug('ondata');
     const ret = dest.write(chunk);
     debug('dest.write', ret);
-    if (ret === false) {
+    if (ret === false && !cleanedUp) {
```

Comment:
Should we call `src.pause()` in case it's cleaned up? If so: that prevents this from happening.

Reply:
> Should we call `src.pause()` in case it's cleaned up? If so: that prevents this from happening.

src.pause() is related to ret === false and has nothing to do with cleanup.